src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -735,3 +735,21 @@ def _jdk_libs_paths_and_digest(self, hermetic_dist):
   @memoized_method
   def _jdk_libs_abs(self, nonhermetic_dist):
     return nonhermetic_dist.find_libs(self._JDK_LIB_NAMES)
+
+  def _double_check_cache_for_vts(self, vts, zinc_compile_context):
+    # Double check the cache before beginning compilation
+    if self.check_cache(vts):
+      self.context.log.debug(f'Snapshotting results for {vts.target.address.spec}')
+      classpath_entry = self._classpath_for_context(zinc_compile_context)
+      assert classpath_entry.directory_digest is None, f'Classpath entry {classpath_entry} from a double-check is expected to *not* have a digest set!'
+      assert os.path.isfile(classpath_entry.path), f'Classpath entry {classpath_entry} from a double-check should point to an existing file!'
+      relpath = fast_relpath(classpath_entry.path, get_buildroot())
+      classes_dir_snapshot, = self.context._scheduler.capture_snapshots([
```

Comment:
Does digest.load read the content of the directory recursively? If so, I was wondering  what savings it might have.

Reply:
Will do!

Reply:
Seeing the implementation https://github.com/pantsbuild/pants/blob/33a6d51db91ea1fe69c39117c8878abcb824cdfb/src/python/pants/engine/fs.py#L100-L111

indeed.

Reply:
> Does digest.load read the content of the directory recursively? If so, I was wondering what savings it might have.

It contains the digest which can just be read as a string, rather than needing to do the expensive disk I/O and CPU of reading and hashing the whole file or directory.